addons/account/tests/test_account_payment_register.py

```diff
@@ -2036,3 +2036,18 @@ def test_branch_user_register_payment(self):
         self.env.company.parent_ids.invalidate_recordset()
         payment = wizard._create_payments()
         self.assertTrue(payment)
+
+    def test_duplicate_bank_partner(self):
+        """Test that duplicate_bank_partner_ids doesn't return res.partner(None,)"""
+
+        # Create test data
+        partner1 = self.env['res.partner'].create({'name': 'Partner 1'})
+
+        # Bank with no duplicates
+        bank_unique = self.env['res.partner.bank'].create({
+            'acc_number': 'UNIQUE123',
+            'partner_id': partner1.id,
+        })
+
+        # Test: Should not be res.partner(None,)
+        self.assertNotEqual(bank_unique.duplicate_bank_partner_ids, self.env['res.partner'].browse([None]))
```

Comment:
It is better to test what it should be than (one value of) what it should not be.
```suggestion
        self.assertFalse(bank_unique.duplicate_bank_partner_ids)
```
 
Please also adjust the the doc string.